### PR TITLE
[FIX] find_and_replace: update search undo/redo & add/delete header

### PR DESCRIPTION
**Pull request by jash-odoo**
## Description:

1. Identified the issue where undo or redo operations would result in the removal of all matches, requiring a fresh search.

2. Noted a similar issue where adding or removing rows/columns did not update the search accordingly.

This PR addresses these issues by updating the search functionality whenever any of the mentioned operations are executed. The search results will be automatically updated to reflect the changes

Task: : [3422484](https://www.odoo.com/web#id=3422484&cids=2&menu_id=4720&action=333&active_id=2328&model=project.task&view_type=form)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_lt("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: tests/plugins/find_and_replace.test.ts [hunk 1]
```diff
@@ -150,6 +156,84 @@ describe("basic search", () => {
...
+    model.dispatch("UPDATE_SEARCH", { toSearch: "1", searchOptions });
+    let matches = model.getters.getSearchMatches();
+    let matchIndex = model.getters.getCurrentSelectedMatchIndex();
+    expect(matches).toHaveLength(4);
+    expect(matchIndex).toStrictEqual(0);
+    expect(matches[0]).toStrictEqual({ col: 0, row: 1, selected: true });
+    expect(matches[1]).toStrictEqual({ col: 0, row: 2, selected: false });
+    expect(matches[2]).toStrictEqual({ col: 0, row: 3, selected: false });
+    expect(matches[3]).toStrictEqual({ col: 0, row: 4, selected: false });
+    model.dispatch("REMOVE_COLUMNS_ROWS", {
```
Comment (rrahir) [inline tests/plugins/find_and_replace.test.ts:170]: `deleteRow` helper

File: tests/plugins/find_and_replace.test.ts [hunk 2]
```diff
@@ -150,6 +156,84 @@ describe("basic search", () => {
...
+    expect(matches[2]).toStrictEqual({ col: 0, row: 3, selected: false });
+  });
+
+  test("Need to update search if doing undo redo operation which can update the cell", () => {
+    model.dispatch("UPDATE_SEARCH", { toSearch: "1", searchOptions });
+    let matches = model.getters.getSearchMatches();
+    let matchIndex = model.getters.getCurrentSelectedMatchIndex();
+    expect(matches).toHaveLength(4);
+    expect(matchIndex).toStrictEqual(0);
+    model.dispatch("REMOVE_COLUMNS_ROWS", {
```
Comment (rrahir) [inline tests/plugins/find_and_replace.test.ts:190]: `deleteRow`

**Comment by rrahir** [comment]
also the commit message  is not complete, it does not mention add/remove col/row

File: tests/plugins/find_and_replace.test.ts (hunk 2 above)
Comment (jash-odoo) [reply tests/plugins/find_and_replace.test.ts]: Changes done and update the commit message

**Comment by rrahir** [comment]
there is a conflict

**Comment by jash-odoo** [comment]
Conflict has been resolved :)

**Comment by rrahir** [comment]
@robodoo r+

**Pull request closed without merging**